overlord/snapstate/handlers.go

```diff
@@ -1931,8 +1932,21 @@ func (m *SnapManager) maybeRestart(t *state.Task, info *snap.Info, rebootInfo bo
 	st := t.State()
 
 	if rebootInfo.RebootRequired {
-		t.Logf("Requested system restart.")
-		RestartSystem(t, &rebootInfo)
+		// Store current boot id to be able to check later if we have
```

Comment:
as we discussed we should review this approach, also there is something not quite nice in terms of separation of concerns with this code vs overlord/restart responsabilities